### PR TITLE
feat: Kyma requeue reason metrics

### PR DESCRIPTION
**Pull request by nesmabadr**
**Description**

Changes proposed in this pull request:

- Add metrics with label for the requeue of Kyma reconciliation

**Related issue(s)**
#1112

### CODE REVIEW

File: internal/controller/kyma_controller.go [hunk 1]
```diff
@@ -131,6 +133,7 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 			r.deleteRemoteClientCache(ctx, kyma)
 			r.enqueueWarningEvent(kyma, syncContextError, err)
```
Comment (ruanxin) [inline internal/controller/kyma_controller.go:134]: before this enqueueWarningEvent should also have a reason, right?

File: internal/controller/kyma_controller.go [hunk 2]
```diff
@@ -371,6 +390,7 @@ func (r *KymaReconciler) handleProcessingState(ctx context.Context, kyma *v1beta
 	}
 
 	if err := errGroup.Wait(); err != nil {
+		r.Metrics.RecordRequeueReason(requeueReason)
```
Comment (ruanxin) [inline internal/controller/kyma_controller.go:393]: this `Metrics.RecordRequeueReason` should not be put here, there has race condition in each errGroup.Go routine, you'll lost some error, for example, the `ManifestReconciliationError` and `ModuleCatalogSyncError` may happens during same time period, however, the err returned here is the first one get returned by those two threads (goroutine), and you will lost another error.

Instead, put RecordRequeueReason(requeueReason) in each errGroup goroutine.

**Review by ruanxin — CHANGES_REQUESTED** [review]
lgtm, have some comments

File: internal/controller/kyma_controller.go (hunk 1 above)
Comment (nesmabadr) [reply internal/controller/kyma_controller.go:134]: This is the same error as the SyncContextRetrievalError, the requeue will happen in line 137

**Review by ruanxin — APPROVED** [review]

**Review by ruanxin — APPROVED** [review]

**Pull request merged**